apps/src/netsim/NetSimEntity.js

```diff
@@ -110,3 +110,25 @@ NetSimEntity.prototype.getTable_ = function () {
 NetSimEntity.prototype.buildRow_ = function () {
   return {};
 };
+
+/**
+ * Destroys all provided entities (from remote storage) asynchronously, and
+ * calls onComplete when all entities have been destroyed and/or an error occurs.
+ * @param {NetSimEntity[]} entities
+ * @param {!NodeStyleCallback} onComplete
+ */
+NetSimEntity.destroyEntities = function (entities, onComplete) {
+  if (entities.length === 0) {
+    onComplete(null, true);
+    return;
+  }
+
+  entities[0].destroy(function (err, result) {
+    if (err) {
+      onComplete(err, result);
+      return;
+    }
+
+    NetSimEntity.destroyEntities(entities.slice(1), onComplete);
```

Comment:
This seems clear enough and it's working, I'd only change if it would cause a (1) much better average-case experience [and likely not significantly higher performance needs on the backend] or (2) less-bad worse edge-case user experience [i.e., would leaving the tail of a list un-deleted be a better/worse state than leaving N/10 items undeleted?]

Seems perfectly fine the way it is to me assuming the user-effect difference is not likely significant. Also something that could be revisited if things crop up in testing